RefApp/ViewController.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  ViewController.swift
+//  RefApp
+//
+//  Created by Karl Holmlöv on 2018-10-30.
+//  Copyright © 2018 emp. All rights reserved.
+//
+
+import UIKit
+import ExposurePlayback
+import Player
+import Exposure
+
+class ViewController: UIViewController {
+
+    @IBOutlet weak var expUrl: UITextField!
+    @IBOutlet weak var customer: UITextField!
+    @IBOutlet weak var username: UITextField!
+    @IBOutlet weak var password: UITextField!
+    @IBOutlet weak var buisnessUnit: UITextField!
+
+    @IBAction func Login(_ sender: Any) {
+        guard let testedUrl = expUrl.text, let testedCustomer = customer.text, let testedBuisnessUnit = buisnessUnit.text, testedUrl != "", testedCustomer != "", testedBuisnessUnit != "" else {
+            let alert = UIAlertController(title: "Invalid env", message: "Provide a valid env", preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: "OK", style: .cancel) { _ in })
+            self.present(alert, animated: true)
+            return
+        }
+        let environment = Environment(baseUrl: testedUrl, customer: testedCustomer, businessUnit: testedBuisnessUnit)
+        
+        UserDefaults.standard.set(expUrl.text!, forKey: "expUrl")
```

Comment:
the above guard statement safely unwraps the optional text: Use those values instead, avoiding the ! operator

UserDefaults.standard.set(testedUrl, forKey: "expUrl")

Same below